File: data/CodeLlama/readable_format_HPEC/fortran/file272.f90
program DRB165_taskdep4_orig_yes_omp_50
    use omp_lib
    implicit none

    !$omp parallel
    !$omp single
    call foo()
    !$omp end single
    !$omp end parallel
contains
    subroutine foo()
        implicit none
        integer :: x, y
        x = 0
        y = 2

        !$omp task depend(inout: x) shared(x)
        x = x+1                                 !!1st Child Task
        !$omp end task

        !$omp task shared(y)
        y = y-1                                 !!2nd child task
        !$omp end task

        !$omp taskwait depend(in: x)            !!1st taskwait

        print*, "x=", x
        print*, "y=", y

        !$omp taskwait                          !!2nd taskwait
    end subroutine foo
end program

File: data/gpt-4o/readable_format_HPEC/cpp/file272.cpp
#include <stdio.h>
#include <omp.h>

void foo();

int main()
{
#pragma omp parallel
 {
#pragma omp single
 foo();
 }
 return 0;
}

void foo()
{
 int x = 0, y = 2;

#pragma omp task shared(x)
 x+=1; // 1st Child Task

#pragma omp task shared(y)
 y-=1; // 2nd Child Task

#pragma omp taskwait // 1st taskwait

 printf ("x=%d\n", x);
 printf ("y=%d\n", y);

#pragma omp taskwait // 2nd taskwait
}
